### DISCUSSION TITLE
Inclusion in WinterCG runtime-keys spec

### DISCUSSION THREAD

**Discussion by Ethan-Arrowood** [react-native-community/discussions-and-proposals#535] [Ideas]
Hey React Native folks! I hope this is the right place to open this - happy to move it somewhere else instead too.

I am working to land a new specification in WinterCG that defines a set of **keys** that represent JavaScript runtimes such that these keys can be used in developer tooling. The best example is the usage within `package.json#exports` field. Since `react-native` is already being used by Webpack as a target environment (https://webpack.js.org/guides/package-exports/#target-environment) I have automatically included the key in the proposal draft. 

Before WinterCG officially publishes this specification I want to make we get _approval_/_sign off_ from the React Native project. Check out your section here: https://github.com/wintercg/runtime-keys/pull/2/files#diff-5e793325cd2bfc452e268a4aa2f02b4024dd9584bd1db3c2595f61f1ecf7b985R46-R54 

If you have any feedback please let me know and I'd be happy to adjust. Happy to answer any additional questions you may have as well. Thank you!

**Comment by huntie** [comment]
@Ethan-Arrowood Thanks for sharing this! This is relevant to our recently shared proposal for how React Native's Metro bundler will support package exports: https://github.com/react-native-community/discussions-and-proposals/pull/534 (which also references that Webpack docs section).

In response to your question: we're happy for `react-native` to be included, with small wording feedback — and the below note. I'll add feedback on the wintercg/runtime-keys PR.

**⚠️ What should this represent for you?**

We're in the process of reviewing exactly what `react-native` should represent when used as a community condition like this, as behaviour has been inconsistent in the past across tools (e.g. bundlers) and we want to give the community a clear definition.

Conceptually, React Native is a framework that targets native **and web** platforms (and also platforms beyond mobile, e.g. macOS and Windows — see [React's Many Platform Vision](https://reactnative.dev/blog/2021/08/26/many-platform-vision)). However we note that the community has previously used `react-native` conditions (e.g. existing top-level package.json field) to refer to **_React Native for native platforms_** only — as you are intending.

**However**, your spec is aiming to categorise **_JavaScript runtimes_**, which we noted exactly [in our `exports` proposal](https://github.com/react-native-community/discussions-and-proposals/pull/534/files#diff-40781647168e437e711b426935003e99ce4bddb4c1a5f1ebd61b0218d2b45670R275) (but may yet pivot from: [[1]]([proposals/0534-metro-package-exports-support.md](https://github.com/react-native-community/discussions-and-proposals/pull/534/files/70df00efbdce50fc533a6d2c2a7c90b7068ade6f#diff-40781647168e437e711b426935003e99ce4bddb4c1a5f1ebd61b0218d2b45670)), [[2]](https://github.com/react-native-community/discussions-and-proposals/pull/534#issuecomment-1285311539)):

> We see a space for `"react-native"`, representing a React Native runtime environment (i.e. Hermes or JavaScript Core VMs), to be introduced — enabling packages to make compatibility or performance decisions around this in select cases.

This is adequate to match your description, however I wonder if you are strictly looking for a JavaScript runtime it may be worth being more precise with two replacement entries:

- `hermes`: Hermes — The JavaScript engine optimised for React Native\
https://hermesengine.dev/
- `jsc`: JavaScriptCore (used in React Native pre-Hermes and in other contexts)\
https://developer.apple.com/documentation/javascriptcore

**What impacts this decision is what your developer audience/understanding is** — this granularity may be unhelpful, in which case perhaps `react-native` (runtimes), as suggested, could include (or alias to) the above.

cc @motiz88 @kelset

**Reply by motiz88** [reply]
@huntie: As (for example) there's going to be a `node` key and not a `v8` key, I think the concept of a "runtime" here maps better to React Native than to JSC/Hermes.

**Reply by Ethan-Arrowood** [reply]
@motiz88 makes a great point. I'd rather we use `react-native` for now. If in the future it seems necessary to introduce `hermes` or others like `v8` we can consider that.

**Comment by Ethan-Arrowood** [comment]
And while yes, `react-native` might not be as similar to other runtimes like Node.js or Deno just yet, since it is a unique platform target like `electron` I definitely want to include it. Thank you for reaching back out so quickly 😄

**Comment by nikelborm** [comment]
Related https://github.com/WinterTC55/runtime-keys/issues/20
